components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java

```diff
@@ -430,7 +438,31 @@ public static APIDTO fromAPItoDTO(ApiTypeWrapper model, String tenantDomain) thr
         } else {
             apidto = fromAPItoDTO(model.getApi(), tenantDomain);
         }
-        apidto.setEndpointURLs(fromAPIRevisionListToEndpointsList(apidto, tenantDomain));
+
+        if (!apidto.isIsAWSAPI()) {
+            apidto.setEndpointURLs(fromAPIRevisionListToEndpointsList(apidto, tenantDomain));
+        } else {
+            //getting the server url from the swagger to be displayed as the endpoint url in the dev portal for aws apis
+            JsonElement configElement = new JsonParser().parse(apidto.getApiDefinition());
+            JsonObject configObject = configElement.getAsJsonObject();  //swaggerDefinition as a json object
+            JsonArray servers = configObject.getAsJsonArray("servers");
+            JsonObject server = servers.get(0).getAsJsonObject();
+            String url = server.get("url").getAsString();
+            JsonObject variables = server.getAsJsonObject("variables");
+            JsonObject basePath = variables.getAsJsonObject("basePath");
+            String stageName = basePath.get("default").getAsString();
+            String hostUrl = url.replace("/{basePath}", stageName);
+            if (hostUrl == null) {
+                hostUrl = " ";
+            }
+            APIEndpointURLsDTO apiEndpointURLsDTO = new APIEndpointURLsDTO();
+            List<APIEndpointURLsDTO> endpointUrls = new ArrayList<>();
+            APIURLsDTO apiurLsDTO = new APIURLsDTO();
+            apiurLsDTO.setHttps(hostUrl);
+            apiEndpointURLsDTO.setUrLs(apiurLsDTO);
+            endpointUrls.add(apiEndpointURLsDTO);
+            apidto.setEndpointURLs(endpointUrls);
+        }
```

Comment:
Shall we move this whole code segment to a separate function named **setEndpointURLsForAwsAPIs** perhaps?

Reply:
ack aiya